### ISSUE TITLE
Mission redirected bug

### ISSUE THREAD

**Issue by Darkcyde13** [EDCD/EDDI#2296]
### Expected
Multiple mission redirections to be ignored.

### Observed
Multiple mission redirections, in quick succession, are all voiced.

### Steps to reproduce
1. Take multiples of the same mission.  In my case, Robigo passenger missions.
2. When a redirect happens, the script appears to fire simultaneously, thus ignoring any change of SetState() variables, and/or doesn't populate `event.name`.

### Configuration
* Version [Please give full version numbers here, not just "latest" or "current"]
  * EDDI Version: v4.0.1
  * VoiceAttack Version: N/A
* Localization [if applicable]
  * EDDI's language: UK English
  * Voice: Ivona Amy

### Investigation Notes
I've been running some Robigo passenger missions to try them out, as I've not tried them before.  I noticed that when scanning the visitor beacon in Sothis, all the mission redirects were being voiced when I didn't think they should be.

Looking over the code in 'Mission redirected', it all looks fine.  I checked the Journal events and they too seem to be correct.  So I made a copy of the default personality and added some lines just before the first IF statement, to check the variables:
```
{set miss to MissionDetails(event.missionid)}
{event.missionid}: {miss.name}.

{if state.eddi_context_last_subject = "mission": 1 |else: {dump(state.eddi_context_last_subject)}}
{if state.eddi_context_last_action = "redirected": 2 |else: {dump(state.eddi_context_last_action)}}
{if state.eddi_context_redirected_mission = event.name: 3 |else: {dump(event.name)}}
{if state.eddi_context_redirected_from = redirected_from : 4 |else: {dump(redirected_from)}}
{if state.eddi_context_redirected_to = redirected_to: 5 |else: {dump(redirected_to)}}
```

When I ran another set of missions, this is the output I got:

> 831588824: Mission_Sightseeing_Terrorist_CIVILWAR. "fsd" "disengage"  "Sothis_" "Robigo_Robigo Mines" Sightseeing mission redirected to Robigo Mines in the Robigo system.
> 831588445: Mission_Sightseeing_Criminal_BOOM. "fsd" "disengage"  "Sothis_" "Robigo_Robigo Mines" Sightseeing mission redirected to Robigo Mines in the Robigo system.
> 831588226: Mission_Sightseeing_Doctor_CIVILWAR. "fsd" "disengage"  "Sothis_" "Robigo_Robigo Mines" Sightseeing mission redirected to Robigo Mines in the Robigo system.
> 831587428: Mission_Sightseeing_Criminal_BOOM. "fsd" "disengage"  "Sothis_" "Robigo_Robigo Mines" Sightseeing mission redirected to Robigo Mines in the Robigo system.
> 831588331: Mission_Sightseeing_Doctor_CIVILWAR. "fsd" "disengage"  "Sothis_" "Robigo_Robigo Mines" Sightseeing mission redirected to Robigo Mines in the Robigo system.
> 831587203: Mission_Sightseeing_Scientist_CIVILWAR. 1 2  4 5 Sightseeing mission redirected to Robigo Mines in the Robigo system.

I think I may have scanned the beacon a bit too quickly after dropping from supercruise, as I don't remember "fsd" "disengage" being reported from my earlier attempts at checking variables, but I'll double check this with some more tests.

In my earlier tests, and as you can see in the last line above, 1 2 4 5 is being spoken.  This shows that the first two, and last two, `state` variables match the event variables.  However, in all my tests `event.name` is always 'void', and as noted recently in another issue, SetState() can't be set to 'void'.  Thus, the first IF check will always fail when missions are redirected almost simultaneously, as `event.name` seems to always be 'void'.

You'll also notice that EDDI voices the output in a different order to that listed in the Journal (see below).  I know this probably shouldn't matter, but there does appear to be some kind of timing issue with the execution of the script.  It seems the first five all happened before EDDI realised the SetState() variables had been updated, and only the last one used the new data.

I'm going to run some more tests, and see if using MissionDetails() to check the `.invariantTags` will work in place of `event.name` as they do seem to be updated with each run of the script.  Well, I'll see what happens...

### EDDI Logs
None.

### Player journals
Section of relevant entries from the Journal:
```
{ "timestamp":"2021-12-28T19:31:55Z", "event":"MissionRedirected", "MissionID":831588226, "Name":"Mission_Sightseeing_Doctor_CIVILWAR", "NewDestinationStation":"Robigo Mines", "NewDestinationSystem":"Robigo", "OldDestinationStation":"", "OldDestinationSystem":"Sothis" }
{ "timestamp":"2021-12-28T19:31:55Z", "event":"MissionRedirected", "MissionID":831588331, "Name":"Mission_Sightseeing_Doctor_CIVILWAR", "NewDestinationStation":"Robigo Mines", "NewDestinationSystem":"Robigo", "OldDestinationStation":"", "OldDestinationSystem":"Sothis" }
{ "timestamp":"2021-12-28T19:31:55Z", "event":"MissionRedirected", "MissionID":831587428, "Name":"Mission_Sightseeing_Criminal_BOOM", "NewDestinationStation":"Robigo Mines", "NewDestinationSystem":"Robigo", "OldDestinationStation":"", "OldDestinationSystem":"Sothis" }
{ "timestamp":"2021-12-28T19:31:55Z", "event":"MissionRedirected", "MissionID":831588445, "Name":"Mission_Sightseeing_Criminal_BOOM", "NewDestinationStation":"Robigo Mines", "NewDestinationSystem":"Robigo", "OldDestinationStation":"", "OldDestinationSystem":"Sothis" }
{ "timestamp":"2021-12-28T19:31:55Z", "event":"MissionRedirected", "MissionID":831588824, "Name":"Mission_Sightseeing_Terrorist_CIVILWAR", "NewDestinationStation":"Robigo Mines", "NewDestinationSystem":"Robigo", "OldDestinationStation":"", "OldDestinationSystem":"Sothis" }
{ "timestamp":"2021-12-28T19:31:55Z", "event":"MissionRedirected", "MissionID":831587203, "Name":"Mission_Sightseeing_Scientist_CIVILWAR", "NewDestinationStation":"Robigo Mines", "NewDestinationSystem":"Robigo", "OldDestinationStation":"", "OldDestinationSystem":"Sothis" }
```

**Comment by Darkcyde13** [comment]
I've made a couple of adjustments to the test code and modified the check for `event.name` to be `miss.invariantTags[0]`, then done another run.  This time I waited about 5 seconds after dropping out of SC before I targeted the beacon to scan it.

This is the updated code:
```
{event.missionid}: {miss.invariantTags[0]}.
```

I also added 'Ignored' to the first IF block, so I could see when everything was working as it should.  This is the result:
> 831626386: Sightseeing. "fsd" "disengage" "" 4 5 Sightseeing mission redirected to Robigo Mines in the Robigo system.
> 831626621: Sightseeing. "fsd" "disengage" "" 4 5 Sightseeing mission redirected to Robigo Mines in the Robigo system.
> 831627225: Sightseeing. 1 2 "" 4 5 Ignored.
> 831626910: Sightseeing. 1 2 "" 4 5 Ignored.
> 831627857: Sightseeing. 1 2 "" 4 5 Ignored.
> 831628009: Sightseeing. 1 2 "" 4 5 Ignored.
> 831628114: Sightseeing. 1 2 "" 4 5 Ignored.
> 831627626: Sightseeing. 1 2 "" 4 5 Ignored.

I'm not sure if waiting those 5 seconds helped, or if this is just a fluke, but either way the second one still should not have been voiced.  The 4 5 in the first two lines shows the data from the last run is still present, as it should be.

So there is definitely something wrong with `event.name` as this seem to always be void, and some kind of timing issue too, as the script runs through one or more times when it should not.

Edit: Just for completions sake, here's the Journal entries from the above.  This time, they do seem to be in the correct order...
```
{ "timestamp":"2021-12-28T21:42:29Z", "event":"MissionRedirected", "MissionID":831626386, "Name":"Mission_Sightseeing_Doctor_CIVILWAR", "NewDestinationStation":"Robigo Mines", "NewDestinationSystem":"Robigo", "OldDestinationStation":"", "OldDestinationSystem":"Sothis" }
{ "timestamp":"2021-12-28T21:42:29Z", "event":"MissionRedirected", "MissionID":831626621, "Name":"Mission_Sightseeing_Criminal_BOOM", "NewDestinationStation":"Robigo Mines", "NewDestinationSystem":"Robigo", "OldDestinationStation":"", "OldDestinationSystem":"Sothis" }
{ "timestamp":"2021-12-28T21:42:29Z", "event":"MissionRedirected", "MissionID":831627225, "Name":"Mission_Sightseeing_Scientist_CIVILWAR", "NewDestinationStation":"Robigo Mines", "NewDestinationSystem":"Robigo", "OldDestinationStation":"", "OldDestinationSystem":"Sothis" }
{ "timestamp":"2021-12-28T21:42:29Z", "event":"MissionRedirected", "MissionID":831626910, "Name":"Mission_Sightseeing_Criminal_BOOM", "NewDestinationStation":"Robigo Mines", "NewDestinationSystem":"Robigo", "OldDestinationStation":"", "OldDestinationSystem":"Sothis" }
{ "timestamp":"2021-12-28T21:42:29Z", "event":"MissionRedirected", "MissionID":831627857, "Name":"Mission_Sightseeing_General_CIVILWAR", "NewDestinationStation":"Robigo Mines", "NewDestinationSystem":"Robigo", "OldDestinationStation":"", "OldDestinationSystem":"Sothis" }
{ "timestamp":"2021-12-28T21:42:29Z", "event":"MissionRedirected", "MissionID":831628009, "Name":"Mission_Sightseeing_Scientist_CIVILWAR", "NewDestinationStation":"Robigo Mines", "NewDestinationSystem":"Robigo", "OldDestinationStation":"", "OldDestinationSystem":"Sothis" }
{ "timestamp":"2021-12-28T21:42:29Z", "event":"MissionRedirected", "MissionID":831628114, "Name":"Mission_Sightseeing_Doctor_CIVILWAR", "NewDestinationStation":"Robigo Mines", "NewDestinationSystem":"Robigo", "OldDestinationStation":"", "OldDestinationSystem":"Sothis" }
{ "timestamp":"2021-12-28T21:42:29Z", "event":"MissionRedirected", "MissionID":831627626, "Name":"Mission_Sightseeing_Doctor_CIVILWAR", "NewDestinationStation":"Robigo Mines", "NewDestinationSystem":"Robigo", "OldDestinationStation":"", "OldDestinationSystem":"Sothis" }
```

**Comment by Tkael** [comment]
Regarding `{event.name}`, I was able to identify a problem there. We expected the journal property to be titled "MissionName" but it is instead titled simply "Name" (this was apparently an undocumented journal change that we had missed). Corrected for the next release. 

I'll add some code to suppress multiple redirects targeting the same new destination system and station for the next release.

**Comment by Tkael** [comment]
{SetState()} isn't very effective for this purpose because speech processing is performed by multiple threads in parallel. This change adds additional suppression prior to speech parallelization.
